### PR TITLE
Auth implementation

### PR DESCRIPTION
**Pull request by MosDevx**
## Changes Implemented
- Add devise gem for authentication
- Set up email confirmation and recovery

# Note:
## Start the server with ./bin/dev

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable, :confirmable
   has_many :posts, dependent: :destroy
   has_many :comments, dependent: :destroy
```
Comment (TadesseDev) [inline app/models/user.rb:7]: - [x] Great job adding the `devise gem` and customizing your layouts Team. But I notice your layout needs some arrangement to look like the given wireframe. please make sure all the form elements are aligned to the center and are equal in length as shown on the wireframe. 
### Given wireframe 
blog_login


### Your design 
image

- [Optional] Also, once the app starts it crashes when I try to reload unless I follow the provided link, This Is not a good idea according to a user experience, Please take a look.

https://github.com/MosDevx/BlogApp/assets/69077061/83edd347-07eb-4d1b-8331-5ad2f0f15dd0

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi Team

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x]  Correct GitFlow  ✅
 - [x]  Good commit history  ✅
 - [x]  Good PR title/description  ✅
 - [x]  All linters passed  ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (MosDevx) [reply app/models/user.rb:7]: Hey @TadesseDev . Thanks for the feedback. 
Unfortunatel I've been unable to recreate this bug.
Did it happen consistently?

Comment (TadesseDev) [reply app/models/user.rb:7]: Hello @MosDevx, yes it keeps happening even after I switch to a new device. Please check with a different computer than the one you used for development. If it works fine by then, Let the second reviewer review this, please.

**Review by addod19 — CHANGES_REQUESTED** [review]
### Hi @MosDevx and @Talha-Hanif5666, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- [ ] Nice job making some of the changes :rocket: 


 ## Required Changes ♻️
_Check the comments under the review._
- [x] I get the following error when I run `rails s`
Screenshot from 2023-05-18 06-30-38
To fix this, kindly rename `application.tailwind.css` to `tailwind.css` that is the error message in the browser as seen
- [ ] Also as mentioned in the previous review, the app is not matched with the given wireframe, let us kindly compare our UI with the wireframe below
Screenshot from 2023-05-18 06-37-47
Screenshot from 2023-05-18 06-36-17
Screenshot from 2023-05-18 06-36-09


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by MosDevx** [comment]
@addod19 Thanks for the review. I think the stylesheet issue is ok now. Could you kindly confirm before I ask for my last review?

**Comment by MosDevx** [comment]
image

@addod19 
With styling the login page looks like that. Is that good enough?

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @MosDevx, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

👍 You did well to make the requested changes.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**